darc/darc.go

```diff
@@ -619,9 +630,35 @@ func EvalExpr(expr expression.Expr, getDarc GetDarc, ids ...string) error {
 
 // evalExprDarc takes an extra visited parameter to track the visited nodes and
 // avoid infinite recursion.
-func evalExprDarc(visited map[string]bool, expr expression.Expr, getDarc GetDarc, acceptDarc bool, ids ...string) error {
+func evalExprDarc(visited map[string]bool, expr expression.Expr, getDarc GetDarc,
+	attrFuncs AttrInterpreters, acceptDarc bool, ids ...string) error {
+
 	var issue error
 	Y := expression.InitParser(func(s string) bool {
+		if strings.HasPrefix(s, "attr") {
+			tokens := strings.SplitN(s, ":", 3)
+			if len(tokens) != 3 {
+				issue = errors.New("attr has an invalid format")
+				return false
+			}
+			if tokens[0] != "attr" {
+				issue = errors.New("first token should be attr")
+				return false
+			}
+
+			var ok bool
+			var attrFunc func(string) error
+			if attrFunc, ok = attrFuncs[tokens[1]]; !ok {
+				issue = errors.New("no such attr interpreter: " + tokens[1])
+				return false
+			}
+			if err := attrFunc(tokens[2]); err != nil {
+				issue = err
+				return false
+			}
+			return true
+		}
```

Comment:
Suggestion: for clarity, the `if` branch could have its contents isolated into its own function :
```
func evalAttr(s string, attrFuncs AttrInterpreters) error {
    tokens := strings.SplitN(s, ":", 3)
    if len(tokens) != 3 {
        return errors.New("attr has an invalid format")
    }
    if tokens[0] != "attr" {
        return errors.New("first token should be attr")
    }

    var ok bool
    var attrFunc func(string) error
    if attrFunc, ok = attrFuncs[tokens[1]]; !ok {
        return errors.New("no such attr interpreter: " + tokens[1])
    }
    return attrFunc(tokens[2])
}
```
and reduced to:
```
if strings.HasPrefix(s, "attr") {
    if err := evalAttr(s, attrFuncs); err != nil {
        issue = err
        return false
    }
   return true
}
```
But that's just because I like short functions. :-)